### DISCUSSION TITLE
Custom hostname when sharing notes

### DISCUSSION THREAD

**Discussion by dimib-tech** [TriliumNext/Trilium#3678] [Q&A] [answered]
To keep things brief, will add more info if it is possible to do this.

Use custom domain when sharing note
Instead of `my-app.fly.dev/share/note` I want my custom domain `my-app.com/share/note`

1) Deployed app via docker on fly.io
2) Added custom domain
(e.g. my-app.fly.dev is the domain assigned by the fly.io team) - I believe this is where I need to change something
(e.g. my-app.com is the custom domain assigned by me)

My app is available both on the my-app.fly.dev domain and on the my-app.com domain, and everything works.

I'm guessing there is a dirty workaround by compiling my own trilium SC, adding the custom hostname option and packaging it into a docker image to run later. But I hope to find a neater feature, maybe it exists somewhere in the trilium configuration files... 
I've looked at the doc's but I think my networking knowledge is currently limiting my abilites to find an answer to this. (Or it's just that idk how trilium deploys the server/finds the hostname)

I'd be happy to help developing this feature, or provide more info if needed.

**Comment by zadam** [comment, marked as answer]
Hello, this is not a trilium feature, but something you need to configure in your reverse proxy or webhost. For Trilium it doesn't matter from which hostname it is served, all the links are relative.

**Reply by dimib-tech** [reply]
To just establish that this is answered
![image](https://user-images.githubusercontent.com/59563446/223633269-28724c2b-9a5b-4308-ad85-6219ebae7fa7.png)
This can be changed into a custom link (As in, the custom domain is shown by default)

> "but something you need to configure in your reverse proxy or webhost"
- My note is available from both links, I believe this is configured?

To specify, my goal is for the new domain to show up in the application, when I share a note.
- I need to configure the reverse proxy/webhost to "tell" trilium which hostname to use?

I just need a pointer in the right direction. I'm gonna do my due diligence afterwards.

**Reply by zadam** [reply]
Hi, sorry, I misunderstood the question then.

The URL shown is composed out of `syncServerHost` and then the `/share/...` path.

If the whole trilium is exposed on `my-app.com`, then you can just change the sync URL here:

![image](https://user-images.githubusercontent.com/617641/223637954-a6627940-6990-4da5-a182-75a527904692.png)

It's true that the sync server host and the share host could be different and this would in fact be a Trilium feature. For now they are expected to be the same.

**Reply by dimib-tech** [reply]
Thank you! I'm using Trilium to share notes with some of my students (Learning the basics of Web Development), and I'm just amazed on how much this app can do.

P.S The fact that my notes are available from anywhere I've got internet is amazing. Thank you for marvelous work!
